Sources/OpenAPIURLSession/URLSessionTransport.swift

```diff
@@ -205,6 +218,69 @@ extension URLSessionTransportError: CustomStringConvertible {
         case .notHTTPResponse(let response):
             return "Received a non-HTTP response, of type: \(String(describing: type(of: response)))"
         case .noResponse(let url): return "Received a nil response for \(url?.absoluteString ?? "<nil URL>")"
+        case .streamingNotSupported: return "Streaming is not supported on this platform"
         }
     }
 }
+
+var debugLoggingEnabled: Bool = false
+func debug(_ items: Any..., separator: String = " ", terminator: String = "\n") {
+    assert(
+        {
+            if debugLoggingEnabled { print(items, separator: separator, terminator: terminator) }
+            return true
+        }()
+    )
+}
+
+extension URLSession {
+    func bufferedRequest(for request: HTTPRequest, baseURL: URL, requestBody: HTTPBody?) async throws -> (
+        HTTPResponse, HTTPBody?
+    ) {
+        var urlRequest = try URLRequest(request, baseURL: baseURL)
+        if let requestBody { urlRequest.httpBody = try await Data(collecting: requestBody, upTo: .max) }
+
+        /// Use `dataTask(with:completionHandler:)` here because `data(for:[delegate:]) async` is only available on
+        /// Darwin platforms newer than our minimum deployment target, and not at all on Linux.
+        let (response, maybeResponseBodyData): (URLResponse, Data?) = try await withCheckedThrowingContinuation {
+            continuation in
+            let task = self.dataTask(with: urlRequest) { [urlRequest] data, response, error in
+                if let error {
+                    continuation.resume(throwing: error)
+                    return
+                }
+                guard let response else {
+                    continuation.resume(throwing: URLSessionTransportError.noResponse(url: urlRequest.url))
+                    return
+                }
+                continuation.resume(with: .success((response, data)))
+            }
+            task.resume()
+        }
+
+        let maybeResponseBody = maybeResponseBodyData.map { data in
+            HTTPBody(data, length: HTTPBody.Length(from: response), iterationBehavior: .multiple)
+        }
+        return (try HTTPResponse(response), maybeResponseBody)
+    }
+}
+
+extension URLSessionTransport.Configuration.Implementation {
+    static var platformSupportsStreaming: Bool {
+        #if canImport(Darwin)
+        guard #available(macOS 12, iOS 15, tvOS 15, watchOS 8, *) else { return false }
+        _ = URLSession.bidirectionalStreamingRequest
```

Comment:
Why is this here?

Reply:
It's here to just to check that the availability guard we used above is the right one. Given this is just a computed boolean property, there's nothing that prevented me getting the version numbers in the guard wrong.